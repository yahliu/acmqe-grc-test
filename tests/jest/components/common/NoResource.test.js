/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
/* Copyright (c) 2020 Red Hat, Inc. */
'use strict'

import React from 'react'
import NoResource from '../../../../src-web/components/common/NoResource'
import renderer from 'react-test-renderer'

describe('NoResource component', () => {
  it('renders as expected', () => {
    const topButton = <div>topButton</div>
    const component = renderer.create(
      <NoResource title='title' detail='detailed description' topButton={topButton}>
        <div className='child'>Test</div>
      </NoResource>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
  it('renders as expected', () => {
    const topButton = <div>topButton</div>
    const component = renderer.create(
      <NoResource
        title='title'
        detail='detailed description'
        topButton={topButton} filterToEmpty={true}>
        <div className='child'>Test</div>
      </NoResource>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
  it('renders as expected', () => {
    const topButton = <div>topButton</div>
    const component = renderer.create(
      <NoResource
        title='title'
        detail='detailed description'
        topButton={topButton}
        titleClassName={'test-title'}
        imgClassName={'test-image'}
        svgName={'test-svg.svg'} alt={'test'}>
        <div className='child'>Test</div>
      </NoResource>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
  it('renders as expected', () => {
    const component = renderer.create(
      <NoResource>
        <div className='child'>Test</div>
      </NoResource>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
  it('renders as expected', () => {
    const component = renderer.create(
      <NoResource
        className={'test-class'}
        titleClassName={'test-title'}
        imgClassName={'test-image'}
        svgName={'test-svg.svg'} alt={'test'}>
        <div className='child'>Test</div>
      </NoResource>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
