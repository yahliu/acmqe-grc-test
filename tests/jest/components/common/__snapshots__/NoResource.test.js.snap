// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoResource component renders as expected 1`] = `
<div>
  <div>
    topButton
  </div>
  <div
    className="no-resource"
  >
    <img
      alt="No resource"
      className="no-resource-icon"
      src="undefined/graphics/EmptyPagePlanet-illus.png"
    />
    <div
      className="no-resource-text-info"
    >
      <div
        className="no-resource-title"
      >
        title
      </div>
      <div
        className="no-resource-detail"
      >
        detailed description
      </div>
    </div>
    <div
      className="child"
    >
      Test
    </div>
  </div>
</div>
`;

exports[`NoResource component renders as expected 2`] = `
<div>
  <div>
    topButton
  </div>
  <div
    className="no-resource"
  >
    <img
      alt="No resource"
      className="no-resource-icon"
      src="undefined/graphics/EmptyPagePlanet-illus.png"
    />
    <div
      className="no-resource-text-info"
    >
      <div
        className="no-resource-title"
      >
        title
      </div>
      <div
        className="no-resource-detail"
      >
        detailed description
      </div>
    </div>
    <div
      className="child"
    >
      Test
    </div>
  </div>
</div>
`;

exports[`NoResource component renders as expected 3`] = `
<div>
  <div>
    topButton
  </div>
  <div
    className="no-resource"
  >
    <img
      alt="test"
      className="test-image"
      src="undefined/graphics/test-svg.svg"
    />
    <div
      className="no-resource-text-info"
    >
      <div
        className="test-title"
      >
        title
      </div>
      <div
        className="no-resource-detail"
      >
        detailed description
      </div>
    </div>
    <div
      className="child"
    >
      Test
    </div>
  </div>
</div>
`;

exports[`NoResource component renders as expected 4`] = `
<div
  className="no-resource"
>
  <img
    alt="No resource"
    className="no-resource-icon"
    src="undefined/graphics/EmptyPagePlanet-illus.png"
  />
</div>
`;

exports[`NoResource component renders as expected 5`] = `
<div
  className="test-class"
>
  <img
    alt="test"
    className="test-image"
    src="undefined/graphics/test-svg.svg"
  />
</div>
`;
